src/freenet/node/updater/NodeUpdateManager.java

```diff
@@ -404,7 +412,7 @@ public void onSuccess(FetchResult result, ClientGetter state) {
 			File temp;
 			FileOutputStream fos = null;
 			try {
-				temp = File.createTempFile(filename, ".tmp", node.getRunDir());
+				temp = File.createTempFile(filename, ".tmp", directory.dir());
```

Comment:
Is it an intended behavior change to use directories other than the run directory for the temporary file? What's the reasoning for that? I was going to ask if that would cause problems for packaged versions, but packaged versions are updated by the package system anyway so that's not a problem.